core/src/validator.rs

```diff
@@ -741,7 +725,10 @@ impl Validator {
             check_poh_speed(&bank_forks.read().unwrap().root_bank(), None)?;
         }
 
-        let hard_forks = bank_forks.read().unwrap().root_bank().hard_forks();
+        let (root_slot, hard_forks) = {
+            let root_bank = bank_forks.read().unwrap().root_bank();
+            (root_bank.slot(), root_bank.hard_forks())
+        };
         let shred_version = compute_shred_version(&genesis_config.hash(), Some(&hard_forks));
```

Comment:
to confirm, after this change the only use of the `--expected-shred-version` flag is for the sanity check 5 lines below? It won't influence any behavior with regards to WFSM / backup & clear / or gossip right?

Reply:
perhaps a separate PR but we should update the help messages to be a bit more descriptive

Reply:
I was referring to:
https://github.com/anza-xyz/agave/blob/1334fb5248390dfdd193feeec5fa8f86763668fc/validator/src/cli.rs#L759
And the one for WFSM:
https://github.com/anza-xyz/agave/blob/1334fb5248390dfdd193feeec5fa8f86763668fc/validator/src/cli.rs#L780-L781
Since they're not very descriptive, but it's pretty minor. I don't expect validators to be setting this without our instruction anyway.

Reply:
Ah gotcha, and yeah, I think a bunch of arg help messages could probably use some work ha. Maybe I'll take a crack at this in near future.